gh-pr-upsert

```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+PYTHONPATH="$(dirname $0)"/src python3 -m gh_pr_upsert "$@"
```

Comment:
This two-line file is what makes gh-pr-upsert a GitHub CLI extension not just a Python package. The way GitHub CLI extensions work is:

1. The repo is supposed to be named like `gh-foo-bar` (I think this is just convention rather than strictly necessary)
2. It must contain a top-level executable script file with the same name as the repo itself

This is a minimal script file that just runs out Python package.

So the minimum GitHub CLI extension would be just a single-file bash or Python script in this `gh-pr-upsert` file and literally no other files in the repo at all. But of course I wanted to use our cookiecutter so I could have a multi-file package with formatting, linting, tests, CI, etc.

We don't actually benefit at all from being a GitHub CLI extension. All it does is enable gh-pr-upsert to be installed via `gh extension install hypothesis/gh-pr-upsert`. It doesn't give us any capabilities at all that a plain Python package wouldn't have. But a command for upserting PRs really seems like it belongs as a GitHub CLI extension.